Build & tests fail with musl target

Tried to build vmm-sys-utils with musl and it fails:

```cargo test --all --target x86_64-unknown-linux-musl
   Compiling libc v0.2.58
   Compiling vmm-sys-util v0.1.0 (/home/local/ANT/fandree/sources/work/rust-vmm/vmm-sys-util)
error[E0308]: mismatched types
   --> src/signal.rs:317:9
    |
316 |     fn pthread_handle(&self) -> pthread_t {
    |                                 --------- expected `*mut libc::c_void` because of return type
317 |         self.as_pthread_t()
    |         ^^^^^^^^^^^^^^^^^^^ expected *-ptr, found u64
    |
    = note: expected type `*mut libc::c_void`
               found type `u64`

error: aborting due to previous error
```

--- a/.buildkite/pipeline.yml
+++ b/.buildkite/pipeline.yml
@@ -8,7 +8,7 @@ steps:
       platform: x86_64.metal
     plugins:
       - docker#v3.0.1:
-          image: "fandree/rust-vmm-dev"
+          image: "rustvmm/dev:v2"
           always-pull: true
 
   - label: "build-gnu-arm"
@@ -20,9 +20,49 @@ steps:
       platform: arm.metal
     plugins:
       - docker#v3.0.1:
-          image: "fandree/rust-vmm-dev"
+          image: "rustvmm/dev:v2"
           always-pull: true
           
+  - label: "build-musl-arm"
+    commands:
+     - cargo build --release --target aarch64-unknown-linux-musl
+    retry:
+      automatic: false
+    agents:
+      platform: arm.metal
+      os: linux
+    plugins:
+      - docker#v3.0.1:
+          image: "rustvmm/dev:v2"
+          always-pull: true
+
+  - label: "unittests-musl-x86"
+    commands:
+     - cargo test --all-features --target x86_64-unknown-linux-musl
+    retry:
+      automatic: false
+    agents:
+      platform: x86_64.metal
+      os: linux
+    plugins:
+      - docker#v3.0.1:
+          image: "rustvmm/dev:v2"
+          always-pull: true
+
+  - label: "unittests-musl-arm"
+    commands:
+     - cargo test --all-features --target aarch64-unknown-linux-musl
+    retry:
+      automatic: false
+    agents:
+      platform: arm.metal
+      os: linux
+    plugins:
+      - docker#v3.0.1:
+          image: "rustvmm/dev:v2"
+          always-pull: true
+
+
   - label: "style"
     command: cargo fmt --all -- --check
     retry:
@@ -31,7 +71,7 @@ steps:
       platform: x86_64.metal
     plugins:
       - docker#v3.0.1:
-          image: "fandree/rust-vmm-dev"
+          image: "rustvmm/dev:v2"
           always-pull: true
   
   - label: "unittests-gnu-x86"
@@ -44,7 +84,7 @@ steps:
     plugins:
       - docker#v3.0.1:
           privileged: true
-          image: "fandree/rust-vmm-dev"
+          image: "rustvmm/dev:v2"
           always-pull: true
           tmpfs: [ "/tmp:exec" ]
           
@@ -58,7 +98,7 @@ steps:
     plugins:
       - docker#v3.0.1:
           privileged: true
-          image: "fandree/rust-vmm-dev"
+          image: "rustvmm/dev:v2"
           always-pull: true
           tmpfs: [ "/tmp:exec" ]
   
@@ -71,5 +111,43 @@ steps:
       platform: x86_64.metal
     plugins:
       - docker#v3.0.1:
-          image: "fandree/rust-vmm-dev"
+          image: "rustvmm/dev:v2"
+          always-pull: true
+
+  - label: "clippy-arm"
+    commands:
+     - cargo clippy --all -- -D warnings
+    retry:
+      automatic: false
+    agents:
+      platform: arm.metal
+    plugins:
+      - docker#v3.0.1:
+          image: "rustvmm/dev:v2"
+          always-pull: true
+
+  - label: "check-warnings-x86"
+    commands:
+      - RUSTFLAGS="-D warnings" cargo check --all-targets
+    retry:
+      automatic: false
+    agents:
+      platform: x86_64.metal
+      os: linux
+    plugins:
+      - docker#v3.0.1:
+          image: "rustvmm/dev:v2"
+          always-pull: true
+
+  - label: "check-warnings-arm"
+    commands:
+      - RUSTFLAGS="-D warnings" cargo check --all-targets
+    retry:
+      automatic: false
+    agents:
+      platform: arm.metal
+      os: linux
+    plugins:
+      - docker#v3.0.1:
+          image: "rustvmm/dev:v2"
           always-pull: true

--- a/.cargo/config
+++ b/.cargo/config
@@ -1,0 +1,2 @@
+[target.aarch64-unknown-linux-musl]
+rustflags = [ "-C", "target-feature=+crt-static", "-C", "link-arg=-lgcc"] 

--- a/src/signal.rs
+++ b/src/signal.rs
@@ -314,7 +314,10 @@ pub unsafe trait Killable {
 // Safe because we fulfill our contract of returning a genuine pthread handle.
 unsafe impl<T> Killable for JoinHandle<T> {
     fn pthread_handle(&self) -> pthread_t {
-        self.as_pthread_t()
+        // JoinHandleExt::as_pthread_t gives c_ulong, convert it to the
+        // type that the libc crate expects
+        assert_eq!(mem::size_of::<pthread_t>(), mem::size_of::<usize>());
+        self.as_pthread_t() as usize as pthread_t
     }
 }
 

